moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -126,6 +126,14 @@ class MoEngageDestination(private val application: Application) : DestinationPlu
         try {
             super.identify(payload)
             Logger.print { "$tag identify(): will try to track $payload" }
+            if (payload.userId.isNotEmpty()) {
+                Logger.print { "$tag identify(): will try to track userId" }
+                MoEAnalyticsHelper.setUniqueId(
+                    application.applicationContext,
+                    payload.userId,
+                    instanceId
+                )
+            }
             val traits = payload.traits
             if (traits.isNotEmpty()) {
                 integrationHelper.trackUserAttribute(removeTraitsWithNullValues(traits), instanceId)
```

Comment:
As we are supporting userId with traits, we might have to track unique id from traits separately (may be at same place for both payload.userId & userTraits userId) otherwise data of 1 user can be tracked with previous user

Reply:
Done